language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverObjectResolver.java

```diff
@@ -0,0 +1,308 @@
+            case FUNCTION:
+                return getHoverObjectForSymbol((FunctionSymbol) symbol);
+            case METHOD:
+                return getHoverObjectForSymbol((MethodSymbol) symbol);
+            case RESOURCE_METHOD:
+                return getHoverObjectForSymbol((ResourceMethodSymbol) symbol);
+            case TYPE_DEFINITION:
+                return getHoverObjectForSymbol((TypeDefinitionSymbol) symbol);
+            case CLASS:
+                return getHoverObjectForSymbol((ClassSymbol) symbol);
+            case VARIABLE:
+                return getHoverObjectForSymbol((VariableSymbol) symbol);
+            case PARAMETER:
+                return getHoverObjectForSymbol((ParameterSymbol) symbol);
+            case TYPE:
+                if (symbol instanceof TypeReferenceTypeSymbol) {
+                    return getHoverObjectForSymbol(((TypeReferenceTypeSymbol) symbol).definition());
+                }
+                return HoverUtil.getHoverObject("");
+            default:
+                return HoverUtil.getDescriptionOnlyHoverObject(symbol);
+        }
+    }
+
+    private Hover getHoverObjectForSymbol(VariableSymbol variableSymbol) {
+        Optional<Documentation> documentation = variableSymbol.documentation();
+        List<String> hoverContent = new ArrayList<>();
+        if (documentation.isPresent() && documentation.get().description().isPresent()) {
+            hoverContent.add(documentation.get().description().get());
+        }
+        TypeSymbol varTypeSymbol = variableSymbol.typeDescriptor();
+        if (varTypeSymbol.typeKind() != TypeDescKind.COMPILATION_ERROR) {
+            String type = varTypeSymbol.signature();
+            String varName = variableSymbol.getName().isPresent() ? " " + variableSymbol.getName().get() : "";
+            String modifiedVariable = HoverUtil.quotedString(type) + CommonUtil.escapeEscapeCharsInIdentifier(varName);
+            hoverContent.add(modifiedVariable);
+        }
+        return HoverUtil.getHoverObject(hoverContent.stream()
+                .collect(Collectors.joining(HoverUtil.getHorizontalSeparator())));
+    }
+
+    private Hover getHoverObjectForSymbol(ParameterSymbol symbol) {
+        String hoverContent = "";
+        TypeSymbol typeSymbol = symbol.typeDescriptor();
+        if (typeSymbol.typeKind() == TypeDescKind.COMPILATION_ERROR) {
+            return HoverUtil.getHoverObject(hoverContent);
+        }
+        String type = typeSymbol.signature();
+        String parameterName = symbol.getName().isPresent() ? " " + symbol.getName().get() : "";
+        hoverContent = HoverUtil.quotedString(type) + CommonUtil.escapeEscapeCharsInIdentifier(parameterName);
+        return HoverUtil.getHoverObject(hoverContent);
+    }
+
+    private Hover getHoverObjectForSymbol(ClassSymbol symbol) {
+        Optional<Documentation> documentation = symbol.documentation();
+        if (documentation.isEmpty()) {
+            return HoverUtil.getHoverObject("");
```

Comment:
Passing an empty string is not explicit. What if we add a method without args?

Reply:
Done